### PR TITLE
[1/6] Venft staking form

### PR DESCRIPTION
**Pull request by jmilldotdev**
## What does this PR do and why?

Builds out the veNft staking form. Adds input components for all of the necessary fields, adds shell components for the modals which the form launches, and adds some dummy data to make this work. I intend to add the data models and transactions in a future PR, so right now this is just for the form components.

There is a carousel component which displays the current NFTs, and i have a placeholder for that for now and will bring it in in a future PR.

Related to https://github.com/jbx-protocol/juice-interface/issues/1416

## Screenshots or screen recordings

image

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Review by tomquirk — CHANGES_REQUESTED** [review]
@jmilldotdev woot!!

Code looks pretty good to me on first look.

I had some UX suggestions and instead of enumerating them, I figured I'd just create a patch (below). I still think there's room for improvement here, but not worth blocking the MVP.

Let me know what you think or if there's context I'm missing 🙏 

Also, should this PR be targeting `main` instead?

<details><summary>Diff</summary>

> Apply by copying the below to clipboard, and using `pbpaste | git apply` locally:

```diff
diff --git a/src/components/veNft/VeNftContent.tsx b/src/components/veNft/VeNftContent.tsx
index 83af0b07b..3d150a4a8 100644
--- a/src/components/veNft/VeNftContent.tsx
+++ b/src/components/veNft/VeNftContent.tsx
@@ -9,6 +9,7 @@ import VeNftStakingForm from 'components/veNft/VeNftStakingForm'
 import VeNftOwnedTokensSection from 'components/veNft/VeNftOwnedTokensSection'
 import VeNftSummaryStatsSection from 'components/veNft/VeNftSummaryStatsSection'
 import { BigNumber } from '@ethersproject/bignumber'
+import { Space } from 'antd'
 
 const VeNftContent = () => {
   const { tokenSymbol, tokenName, projectMetadata } =
@@ -24,7 +25,7 @@ const VeNftContent = () => {
   ]
 
   return (
-    <>
+    <Space direction="vertical">
       <VeNftHeaderSection
         tokenName={tokenName}
         tokenSymbolDisplayText={tokenSymbolDisplayText}
@@ -38,7 +39,7 @@ const VeNftContent = () => {
       <VeNftSummaryStatsSection
         tokenSymbolDisplayText={tokenSymbolDisplayText}
       />
-    </>
+    </Space>
   )
 }
 
diff --git a/src/components/veNft/VeNftStakingForm.tsx b/src/components/veNft/VeNftStakingForm.tsx
index 671adcb82..ba2286e88 100644
--- a/src/components/veNft/VeNftStakingForm.tsx
+++ b/src/components/veNft/VeNftStakingForm.tsx
@@ -19,8 +19,6 @@ import VeNftCarousel from 'components/veNft/veNftCarousel'
 import StakingTokenRangesModal from 'components/veNft/veNftStakingTokenRangesModal'
 import ConfirmStakeModal from 'components/veNft/veNftConfirmStakeModal'
 
-import Callout from 'components/Callout'
-
 import { shadowCard } from 'constants/styles/shadowCard'
 
 interface StakingFormProps {
@@ -96,46 +94,43 @@ const VeNftStakingForm = ({
         form={form}
         initialValues={initialValues}
       >
-        <div style={{ ...shadowCard(theme), padding: 25, marginBottom: 10 }}>
-          <Space size="middle" direction="vertical">
-            <Callout>
-              <Trans>
-                Only project tokens claimed as ERC-20 tokens can be staked for
-                NFTs.
-              </Trans>
-            </Callout>
-          </Space>
-          <TokensStakedInput
-            form={form}
-            claimedBalance={claimedBalance}
-            tokenSymbolDisplayText={tokenSymbolDisplayText}
-            tokensStaked={tokensStaked}
-            minTokensAllowedToStake={minTokensAllowedToStake}
-          />
-          <Row>
-            <Col span={14}>
-              <LockDurationSelectInput
-                form={form}
-                lockDurationOptionsInSeconds={lockDurationOptionsInSeconds}
-              />
-            </Col>
-            <Col span={4}>
-              <p style={{ textAlign: 'center' }}>=</p>
-            </Col>
-            <Col span={6}>
+        <div style={{ ...shadowCard(theme), padding: 25 }}>
+          <Space direction="vertical" size={'large'} style={{ width: '100%' }}>
+            <div>
+              <Row gutter={20}>
+                <Col span={14}>
+                  <TokensStakedInput
+                    form={form}
+                    claimedBalance={claimedBalance}
+                    tokenSymbolDisplayText={tokenSymbolDisplayText}
+                    tokensStaked={tokensStaked}
+                    minTokensAllowedToStake={minTokensAllowedToStake}
+                  />
+                </Col>
+                <Col span={10}>
+                  <LockDurationSelectInput
+                    form={form}
+                    lockDurationOptionsInSeconds={lockDurationOptionsInSeconds}
+                  />
+                </Col>
+              </Row>
+
               <VotingPowerDisplayInput votingPower={votingPower} />
-            </Col>
-          </Row>
-          <CustomBeneficiaryInput form={form} />
-          <VeNftCarousel />
-          <Space size="middle" direction="vertical" style={{ width: '100%' }}>
-            <Button block onClick={() => setTokenRangesModalVisible(true)}>
-              <Trans>View Token Ranges</Trans>
-            </Button>
-            <StakingFormActionButton
-              hasAdequateApproval={hasAdequateApproval}
-              onReviewButtonClick={handleReviewButtonClick}
-            />
+            </div>
+
+            <CustomBeneficiaryInput form={form} />
+
+            <VeNftCarousel />
+
+            <Space size="middle" direction="vertical" style={{ width: '100%' }}>
+              <Button block onClick={() => setTokenRangesModalVisible(true)}>
+                <Trans>View token ranges</Trans>
+              </Button>
+              <StakingFormActionButton
+                hasAdequateApproval={hasAdequateApproval}
+                onReviewButtonClick={handleReviewButtonClick}
+              />
+            </Space>
           </Space>
         </div>
       </Form>
diff --git a/src/components/veNft/formControls/CustomBeneficiaryInput.tsx b/src/components/veNft/formControls/CustomBeneficiaryInput.tsx
index 87a88ac30..318010f75 100644
--- a/src/components/veNft/formControls/CustomBeneficiaryInput.tsx
+++ b/src/components/veNft/formControls/CustomBeneficiaryInput.tsx
@@ -1,5 +1,5 @@
 import { t, Trans } from '@lingui/macro'
-import { Form, FormInstance, Switch } from 'antd'
+import { Form, FormInstance, Space, Switch } from 'antd'
 import { EthAddressInput } from 'components/inputs/EthAddressInput'
 import { ThemeContext } from 'contexts/themeContext'
 import { isAddress } from 'ethers/lib/utils'
@@ -28,31 +28,20 @@ const CustomBeneficiaryInput = ({ form }: CustomBeneficiaryInputProps) => {
   }
 
   return (
-    <>
-      <Form.Item
-        label={
-          <>
-            <Trans>Custom token beneficiary</Trans>
-            <Switch
-              checked={customBeneficiaryEnabled}
-              onChange={setCustomBeneficiaryEnabled}
-              style={{ marginLeft: 10 }}
-            />
-          </>
-        }
-        extra={
-          <div style={{ color: colors.text.primary, marginBottom: 10 }}>
-            <p>
-              <Trans>Mint NFT to a custom address</Trans>
-            </p>
-          </div>
-        }
-        style={{ marginBottom: '1rem' }}
-      />
+    <Form.Item extra={t`Mint NFT to a custom address.`}>
+      <Space>
+        <Switch
+          checked={customBeneficiaryEnabled}
+          onChange={setCustomBeneficiaryEnabled}
+        />
+        <span style={{ color: colors.text.primary, fontWeight: 500 }}>
+          <Trans>Custom token beneficiary</Trans>
+        </span>
+      </Space>
       {customBeneficiaryEnabled && (
         <Form.Item
+          style={{ marginTop: '1rem', marginBottom: 0 }}
           name="beneficiary"
-          label="Beneficiary"
           rules={[
             {
               validator: validateCustomBeneficiary,
@@ -62,7 +51,7 @@ const CustomBeneficiaryInput = ({ form }: CustomBeneficiaryInputProps) => {
           <EthAddressInput />
         </Form.Item>
       )}
-    </>
+    </Form.Item>
   )
 }
 
diff --git a/src/components/veNft/formControls/LockDurationSelectInput.tsx b/src/components/veNft/formControls/LockDurationSelectInput.tsx
index a65b38be6..5f0110e66 100644
--- a/src/components/veNft/formControls/LockDurationSelectInput.tsx
+++ b/src/components/veNft/formControls/LockDurationSelectInput.tsx
@@ -1,8 +1,6 @@
 import { Trans } from '@lingui/macro'
 import { Form, Select } from 'antd'
 import { FormInstance } from 'antd/lib/form'
-import { ThemeContext } from 'contexts/themeContext'
-import { useContext } from 'react'
 import { detailedTimeString } from 'utils/formatTime'
 
 interface TokensStakedInputProps {
@@ -14,24 +12,8 @@ const LockDurationSelectInput = ({
   form,
   lockDurationOptionsInSeconds,
 }: TokensStakedInputProps) => {
-  const {
-    theme: { colors },
-  } = useContext(ThemeContext)
-
   return (
-    <Form.Item
-      name="lockDuration"
-      extra={
-        <div
-          style={{
-            color: colors.text.primary,
-            marginBottom: 10,
-          }}
-        >
-          <Trans>Days Locked</Trans>
-        </div>
-      }
-    >
+    <Form.Item name="lockDuration" label={<Trans>Lock duration</Trans>}>
       <Select onChange={val => form.setFieldsValue({ lockDuration: val })}>
         {lockDurationOptionsInSeconds.map((duration: number) => {
           return (
diff --git a/src/components/veNft/formControls/StakingFormActionButton.tsx b/src/components/veNft/formControls/StakingFormActionButton.tsx
index c92e319f0..b104ece8e 100644
--- a/src/components/veNft/formControls/StakingFormActionButton.tsx
+++ b/src/components/veNft/formControls/StakingFormActionButton.tsx
@@ -41,8 +41,9 @@ const StakingFormActionButton = ({
           block
           style={{ whiteSpace: 'pre' }}
           onClick={onReviewButtonClick}
+          type="primary"
         >
-          Review and Confirm Stake
+          Review and confirm stake
         </Button>
       )
     }
diff --git a/src/components/veNft/formControls/TokensStakedInput.tsx b/src/components/veNft/formControls/TokensStakedInput.tsx
index e5103b4d6..e8c0929c2 100644
--- a/src/components/veNft/formControls/TokensStakedInput.tsx
+++ b/src/components/veNft/formControls/TokensStakedInput.tsx
@@ -1,11 +1,10 @@
-import { t, Trans } from '@lingui/macro'
+import { plural, t, Trans } from '@lingui/macro'
 import { Form } from 'antd'
 import FormattedNumberInput from 'components/inputs/FormattedNumberInput'
 import { BigNumber } from '@ethersproject/bignumber'
 import { FormInstance } from 'rc-field-form'
-import { useContext, useEffect } from 'react'
+import { useEffect } from 'react'
 import { fromWad } from 'utils/formatNumber'
-import { ThemeContext } from 'contexts/themeContext'
 
 interface TokensStakedInputProps {
   form: FormInstance
@@ -22,10 +21,6 @@ const TokensStakedInput = ({
   tokensStaked,
   minTokensAllowedToStake,
 }: TokensStakedInputProps) => {
-  const {
-    theme: { colors },
-  } = useContext(ThemeContext)
-
   const totalBalanceInWad = fromWad(claimedBalance)
   const unstakedTokens = claimedBalance
     ? parseInt(totalBalanceInWad) - parseInt(tokensStaked)
@@ -40,7 +35,10 @@ const TokensStakedInput = ({
     const tokensStaked = parseInt(form.getFieldValue('tokensStaked'))
     if (tokensStaked < minTokensAllowedToStake) {
       return Promise.reject(
-        t`You must stake at least ${minTokensAllowedToStake} tokens.`,
+        plural(minTokensAllowedToStake, {
+          one: 'You must stake at least # token.',
+          other: 'You must stake at least # tokens.',
+        }),
       )
     }
     if (tokensStaked > parseInt(totalBalanceInWad)) {
@@ -52,20 +50,21 @@ const TokensStakedInput = ({
   return (
     <Form.Item
       name="tokensStaked"
+      label={<Trans>{tokenSymbolDisplayText} to lock</Trans>}
       rules={[
         {
           validator: validateTokensStaked,
         },
       ]}
       extra={
-        <div style={{ color: colors.text.primary, marginBottom: 10 }}>
-          <p style={{ float: 'left' }}>
-            <Trans>{tokenSymbolDisplayText} to lock</Trans>
-          </p>
-          <p style={{ float: 'right' }}>
-            <Trans>Remaining: {unstakedTokens}</Trans>
-          </p>
-        </div>
+        <Trans>
+          {unstakedTokens} {tokenSymbolDisplayText} remaining
+        </Trans>
+      }
+      tooltip={
+        <Trans>
+          Only project tokens claimed as ERC-20 tokens can be staked for NFTs.
+        </Trans>
       }
     >
       <FormattedNumberInput
@@ -74,6 +73,7 @@ const TokensStakedInput = ({
         onChange={val => {
           form.setFieldsValue({ tokensStaked: val })
         }}
+        suffix={tokenSymbolDisplayText}
       />
     </Form.Item>
   )
diff --git a/src/components/veNft/formControls/VotingPowerDisplayInput.tsx b/src/components/veNft/formControls/VotingPowerDisplayInput.tsx
index bafcdee9d..f46a390ea 100644
--- a/src/components/veNft/formControls/VotingPowerDisplayInput.tsx
+++ b/src/components/veNft/formControls/VotingPowerDisplayInput.tsx
@@ -1,4 +1,4 @@
-import { Trans } from '@lingui/macro'
+import { t, Trans } from '@lingui/macro'
 import { Form, Input } from 'antd'
 import { ThemeContext } from 'contexts/themeContext'
 import { useContext } from 'react'
@@ -16,19 +16,13 @@ const VotingPowerDisplayInput = ({
 
   return (
     <Form.Item
-      extra={
-        <div
-          style={{
-            color: colors.text.primary,
-            marginBottom: 10,
-            textAlign: 'right',
-          }}
-        >
-          <Trans>Voting Power</Trans>
-        </div>
+      label={
+        <h3 style={{ marginBottom: 0, color: colors.text.primary }}>
+          <Trans>Your voting power</Trans>
+        </h3>
       }
     >
-      <Input disabled={true} value={`${votingPower} VotePWR`} />
+      <Input disabled value={votingPower} suffix={t`VotePWR`} />
     </Form.Item>
   )
 }
diff --git a/src/styles/antd-overrides/input.scss b/src/styles/antd-overrides/input.scss
index 66ac27683..92f636400 100644
--- a/src/styles/antd-overrides/input.scss
+++ b/src/styles/antd-overrides/input.scss
@@ -329,3 +329,7 @@ textarea.ant-input::placeholder {
 .ant-input-number-disabled input {
   color: var(--text-tertiary) !important;
 }
+
+.ant-form-item-label > label .ant-form-item-tooltip {
+  color: var(--text-secondary);
+}
\ No newline at end of file

```

</details>

File: src/components/veNft/formControls/StakingFormActionButton.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const approve = () => {
+    return
+  }
+
+  const renderActionButton = () => {
+    if (!userAddress && onSelectWallet) {
+      return (
+        <Button
+          block
+          style={{ whiteSpace: 'pre' }}
```
Comment (tomquirk) [inline src/components/veNft/formControls/StakingFormActionButton.tsx:25]: **question** what's the reasoning for this style?

File: src/components/veNft/formControls/StakingFormActionButton.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  hasAdequateApproval,
+  onReviewButtonClick,
+}: StakingFormActionButtonProps) => {
+  const { userAddress, onSelectWallet } = useContext(NetworkContext)
+
+  const approve = () => {
+    return
+  }
+
+  const renderActionButton = () => {
```
Comment (tomquirk) [inline src/components/veNft/formControls/StakingFormActionButton.tsx:20]: **praise** nice use of render fn

File: src/components/veNft/VeNftContent.tsx [hunk 3]
```diff
@@ -16,14 +17,23 @@ const VeNftContent = () => {
   const tokenSymbolDisplayText = tokenSymbolText({ tokenSymbol })
   const projectName = projectMetadata?.name ?? t`Unknown Project`
 
+  const lockDurationOptions = [
+    BigNumber.from(1),
+    BigNumber.from(7),
+    BigNumber.from(30),
+  ]
```
Comment (tomquirk) [inline src/components/veNft/VeNftContent.tsx:24]: **question**: Is this always gonna be static?

File: src/components/veNft/VeNftStakingForm.tsx [hunk 4]
```diff
@@ -1,29 +1,154 @@
...
+  const { tokenAddress } = useContext(V2ProjectContext)
   const { theme } = useContext(ThemeContext)
 
+  const [form] = useForm<StakingFormProps>()
+
+  const [tokenRangesModalVisible, setTokenRangesModalVisible] = useState(false)
+  const [confirmStakeModalVisible, setConfirmStakeModalVisible] =
+    useState(false)
+
+  const tokensStaked = useWatch('tokensStaked', form) || '1'
```
Comment (tomquirk) [inline src/components/veNft/VeNftStakingForm.tsx:51]: **praise** BULLISH (usewatch!)

File: src/components/veNft/VeNftContent.tsx (hunk 3 above)
Comment (jmilldotdev) [reply src/components/veNft/VeNftContent.tsx:24]: No, this is read from contract starting in PR3

File: src/components/veNft/formControls/StakingFormActionButton.tsx (hunk 1 above)
Comment (jmilldotdev) [reply src/components/veNft/formControls/StakingFormActionButton.tsx:25]: Tested, unnecessary, removed.

**Comment by jmilldotdev** [comment]
Addressed everything @tomquirk . Do you want me to target main here instead?

**Review by tomquirk — COMMENTED** [review]
@jmilldotdev awesome thank you for making those changes.

Consider this comment my approval ✅ . I will leave gh-unapproved for the moment, while I review the other PRs 🙏

**Comment by tomquirk** [comment]
closing in favor of https://github.com/jbx-protocol/juice-interface/pull/1540

**Pull request closed without merging**